youtube_dl/extractor/brainpop.py

```diff
@@ -0,0 +1,66 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+
+class BrainPOPIE(InfoExtractor):
+    _VALID_URL = r'https?:\/\/(?:(.+)\.)?brainpop\.com\/(?P<id>[^\r\n]+)'
+    _TEST = {
+        'url': 'https://www.brainpop.com/english/freemovies/williamshakespeare/',
+        'md5': '676d936271b628dc05e4cec377751919',
+        'info_dict': {
+            'id': 'english/freemovies/williamshakespeare/',
+            'ext': 'mp4',
+            'title': 'William Shakespeare - BrainPOP',
+            'thumbnail': 're:^https?://.*\.png$',
+            'description': 'He could do comedies, tragedies, histories and poetry.  Learn about the greatest playwright in the history of the English language!',
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        self.report_extraction(video_id)
+
+        ec_token = self._html_search_regex(r"ec_token : '([^']*)'", webpage, 'token')
+
+        settings = self._parse_json(self._html_search_regex(r'var settings = ([^;]*)', webpage, 'settings'), video_id)
+        title = settings['title']
+        description = settings['description']
```

Comment:
It's better to return `title` and `description` directly since you are not using the variables anywhere else. Also, description extraction should be non-fatal